### DISCUSSION TITLE
umi4 线上代理本地静态资源的配置问题

### DISCUSSION THREAD

**Discussion by lyj2209914083** [umijs/umi#11777] [Q&A]
在开发项目过程中希望将本地静态资源代理到线上，正常情况下需要在npm命令中指定SOCKET_SERVER地址，如图所示，但是由于项目运行的端口号是变化的，比如可能端口是8001，此时就需要手动修改：SOCKET_SERVER=http://127.0.0.1:8001。希望用户在运行项目只需要npm run dev即可，不需要关注当前的端口号是多少，再去指定SOCKET_SERVER地址，请问有什么办法吗？
<img width="817" alt="image" src="https://github.com/umijs/umi/assets/70557762/4f2145cd-5c7f-4438-8473-b2b8a201d468">

**Comment by fz6m** [comment]
`SOCKET_SERVER` 是 hmr 服务器的位置，和静态资源无关，99% 的情况用不到这个环境变量。

如果你需要本地调试线上，api 应该配置 proxy 或者判断环境变量分流不同的 url 直接代理到线上接口，静态资源没必要代理线上的，因为本地的代码只可能比线上多不可能少，如果你需要特定去验证某个线上资源可能存在问题，就在电脑的网络请求上用 charles 等工具定点代理某个 url 或者某个规则匹配的的 js 、css 文件，达到代理线上资源的目的，比如访问 `http://localhost:8888/umi.js` 代理到 `http://domain.com/umi.js` 。

**Reply by lyj2209914083** [reply]
我确实是使用了对应的代理工具代理某个 url 或者某个规则匹配的的 js 、css 文件，用以排查问题，这个代理操作在我这经常用到，然后代理是需要制定hmr服务器的地址，原因如图，因此希望在运行项目时自动指定hmr服务器的地址
<img width="1420" alt="image" src="https://github.com/umijs/umi/assets/70557762/05d349aa-de16-40aa-bed7-d3717ebaadbb">

**Reply by fz6m** [reply]
这种情况你可以自己加一个本地 zsh alias ，或者就写在 `package.json#scripts` 里，或者写一个插件来提供调试之类的命令。
